.github/workflows/observe-lab.yml

```diff
@@ -0,0 +1,366 @@
+name: Observe Lab
+
+on:
+  workflow_call:
+    inputs:
+      lab-matrix:
+        description: "Lab jobs to observe"
+        required: true
+        type: string
+      project-callee:
+        description: "The callee to dispatch workflows to"
+        required: true
+        type: string
+      sha:
+        description: "commit sha of the project being tested"
+        required: true
+        type: string
+      ref:
+        description: "branch of the project being tested"
+        required: true
+        type: string
+      pr:
+        description: "pull request number of the project being tested"
+        required: true
+        type: string
+    secrets:
+      NETPERF_SYNCER_SECRET:
+        required: true
+jobs:
+  lab-observer:
+    name: Oracle
+    runs-on: windows-latest
+    steps:
+    - name: Observing Lab State...
+      run: |
```

Comment:
This is a lot of code. I think it'd be better to be put in a script file instead of directly in yaml. A good threshold is probably about 10 lines. Less than that can stay in yaml.

Reply:
The main reason why I chose this design was to avoid having to check out the repo on the physical lab machines. 
I wanted to avoid having any cleanup steps in the workflows on the physical lab machines.

In terms of trade-offs here, doing it this way worsens the readability of the code. What are your thoughts on what weighs more here?

Reply:
Ok. That makes sense. Thanks.